src/components/TransformWorkflow/FlowEditor/Components/OperationPanel/Forms/CaseWhenOpForm.tsx

```diff
@@ -149,32 +146,43 @@ const ClauseOperands = ({
                 <Controller
                   control={control}
                   name={`clauses.${clauseIndex}.operands.${operandIndex}.col_val`}
-                  rules={{ required: data.advanceFilter === 'no' }}
-                  render={({ field }) => (
+                  rules={{
+                    required:
+                      data.advanceFilter === 'no' && 'Column is required',
+                  }}
+                  render={({ field, fieldState }) => (
                     <Autocomplete
+                      {...field}
+                      helperText={fieldState.error?.message}
+                      error={!!fieldState.error}
                       options={data.srcColumns.sort((a, b) =>
                         a.localeCompare(b)
                       )}
                       disabled={disableFields}
-                      value={field.value}
-                      onChange={(e, data) => {
-                        field.onChange(data);
-                      }}
                       placeholder="Select column"
                       fieldStyle="transformation"
                     />
                   )}
                 />
               ) : (
-                <Input
-                  label=""
-                  fieldStyle="transformation"
+                <Controller
+                  control={control}
                   name={`clauses.${clauseIndex}.operands.${operandIndex}.const_val`}
-                  register={register}
-                  sx={{ padding: '0' }}
-                  placeholder="Enter the value"
-                  disabled={disableFields}
-                  required={data.advanceFilter === 'no'}
+                  rules={{
+                    required:
+                      data.advanceFilter === 'no' && 'Value is required',
+                  }}
+                  render={({ field, fieldState }) => (
+                    <Input
+                      helperText={fieldState.error?.message}
+                      error={!!fieldState.error}
+                      fieldStyle="transformation"
+                      sx={{ padding: '0' }}
+                      placeholder="Enter the value"
+                      disabled={disableFields}
+                      {...field}
+                    />
+                  )}
```

Comment:
Switching between the radio buttons doesn't work as expected.

- Click the radio button `Value`
- Enter some value
- Click the radio button `Column`
- You will still see the value in `Value` being filled in the `Column` autocomplete. 

Looks like switching between radio buttons input is not working as expected. It works fine in staging i.e. main branch. 

Problem seesm to be with the radio button's `Value` component. It was first using `register` which works fine but now its being wrapped in a Controller which somehow not registering the input value

Same problem is also there in the `Arithmetic` and `WhereFilter` forms where there is a radio buttton asking between a column and a value.

Reply:
This is fixed now. Please check